pkg/hub/submarineragent/controller.go

```diff
@@ -302,3 +336,76 @@ func (c *submarinerAgentController) removeClusterRBACFiles(ctx context.Context,
 		clusterRBACFiles...,
 	)
 }
+
+type SubmarinerConfig struct {
+	NATEnabled             bool
+	BrokerAPIServer        string
+	BrokerNamespace        string
+	BrokerToken            string
+	BrokerCA               string
+	IPSecPSK               string
+	ClusterName            string
+	ClusterCIDR            string
+	ServiceCIDR            string
+	Repository             string
+	Version                string
+	CatalogChannel         string
+	CatalogName            string
+	CatalogSource          string
+	CatalogSourceNamespace string
+	CatalogStartingCSV     string
+}
+
+func newSubmarinerConfig(
+	client kubernetes.Interface,
+	dynamicClient dynamic.Interface,
+	clusterClient clientset.Interface,
+	clusterName, brokeNamespace string) (*SubmarinerConfig, error) {
+	config := &SubmarinerConfig{
+		Repository:             helpers.GetSubmarinerRepository(),
+		Version:                helpers.GetSubmarinerVersion(),
+		NATEnabled:             false,
+		BrokerNamespace:        brokeNamespace,
+		ClusterName:            clusterName,
+		CatalogName:            "submariner-test", //TODO: change this to `submariner`
+		CatalogChannel:         "alpha",
+		CatalogSource:          "example-catalog", //TODO: change this to community catalog
+		CatalogSourceNamespace: "olm",
+		CatalogStartingCSV:     "submariner.v0.8.0", //TODO: get this from version
+	}
+
+	apiServer, err := helpers.GetBrokerAPIServer(dynamicClient)
+	if err != nil {
+		return nil, err
+	}
+	config.BrokerAPIServer = apiServer
+
+	IPSecPSK, err := helpers.GetIPSecPSK(client, brokeNamespace)
+	if err != nil {
+		return nil, err
+	}
+	config.IPSecPSK = IPSecPSK
+
+	token, ca, err := helpers.GetBrokerTokenAndCA(client, brokeNamespace, clusterName)
+	if err != nil {
+		return nil, err
+	}
+	config.BrokerCA = ca
+	config.BrokerToken = token
+
+	clusterType, err := helpers.GetClusterType(clusterClient, config.ClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the cluster type %+v : %+v", config.ClusterName, err)
+	}
+	switch clusterType {
+	case helpers.ClusterTypeOCP:
+		//TODO: the scc should be handled in submariner v0.8.0, but need verify
+		config.NATEnabled = true
+		if version.IsDownStream() {
+			config.CatalogSource = "redhat-operators"
```

Comment:
I  think we only  need to set different catalog source when it is in downstream. We may not set an explicit env to say it is downstream

Reply:
good suggestion